plutus-contract/src/Language/Plutus/Contract/Types.hs

```diff
@@ -294,21 +305,24 @@ mkResult (((initialRes, cpKey), cpStore), newLogs) =
                 fmap (>>= getResult) initialRes
             , _logs = newLogs
             , _checkpointStore = cpStore
+            , _observableState = oldW <> w
             }
       , _continuations = either (const Nothing) id initialRes
       , _checkpointKey = cpKey
       }
 
 runSuspContractEffects ::
-  forall e i a.
-  CheckpointKey
+  forall w e i a.
+  Monoid w
+  => CheckpointKey
   -> CheckpointStore
-  -> Eff (SuspendedContractEffects e i) a
-  -> (((Either e a, CheckpointKey), CheckpointStore), Seq (LogMessage Value))
+  -> Eff (SuspendedContractEffects w e i) a
+  -> ((((Either e a, CheckpointKey), CheckpointStore), w), Seq (LogMessage Value)) -- FIXME
```

Comment:
fixme?

Reply:
Ah yeah I wanted to make that nested tuple a bit nicer.

Reply:
Will fix it.